backend/coreapp/models.py

```diff
@@ -83,5 +102,32 @@ def __hash__(self):
             self.owner,
         ))
 
+    def save(self, *args, **kwargs):
+        # Scratches cannot be owned if they are a root function in a project
+        if self.owner is not None and ProjectFunction.objects.filter(scratch=self).exists():
+            self.owner = None
```

Comment:
I'm a bit confused about this - if a person tries to claim ownership of a base scratch for a project function, I think that should be an error, not just reject the new owner

Reply:
It does reject -- see the `claim` method on ScratchViewSet.

The reason this is here is because adding a project function in the admin interface or w/e should also set owner to None

Reply:
I didn't mean the claim endpoint , I meant just adding a non-null owner for a scratch that's a base function. I think in all cases, that should be an error and not just silently ignore the specified owner

Reply:
So you think this should raise an error rather than just set it to None?

Reply:
I'm thinking about it, and I think this is fine but we should probably also be catching this in the viewset function for create and just not allow the owner to be explicitly set on create...? since the backend sets the owner itself via the profile, right?